### ISSUE TITLE
Apply ReturnValueIgnored to common JDK methods

### ISSUE THREAD

**Issue by gaul** [google/error-prone#810]
I would like `CheckReturnValue` to apply to common JDK methods like `Integer.compareTo(Integer)`, `String.substring(int, int)`, and `Collection.get`.  Could this checker either trigger on these automatically or allow some kind of configuration?

**Comment by Stephan202** [comment]
Thinking big (maybe _too_ big): support for external API annotations, such as [Checker framework's JDK annotations](https://github.com/typetools/checker-framework/tree/master/checker/jdk).

**Comment by eaftan** [comment]
`ReturnValueIgnored` is the JDK version of this check.  It does trigger on `String.substring(int, int)` but not the other two examples.  The intent of the check was to look for places where people think the instance is going to be mutated, e.g. `String.substring`, but in fact it is not.  

`Integer.compareTo(Integer)` and `List.get(int)` don't fall into that bucket, and it seems unlikely to me that people are commonly making the mistake of calling those methods and discarding the return value.  Have you seen instances of that in practice?

You could try editing https://github.com/google/error-prone/blob/master/core/src/main/java/com/google/errorprone/bugpatterns/ReturnValueIgnored.java to flag those additional methods and run it over your codebase to see if anyone makes that mistake.

**Comment by gaul** [comment]
Thanks for suggesting `ReturnValueIgnored`!  I will investigate this further.

I recently encountered an instance of not using `Integer.compareTo` which Coverity flagged:

https://github.com/jclouds/jclouds/commit/46759f8bda00f86ef934345846e22e2bd2b0d7ae

I agree that these errors are likely more common in certain situations.  I have seen several incorrect `equals` and `compareTo` implementations tend to lack any kind of testing.

**Comment by gaul** [comment]
I added a few extra methods to check here:

https://github.com/gaul/error-prone/tree/return-value-ignored-extras

Not sure if this has enough utility to consider.

**Comment by cpovirk** [comment]
FWIW, when we added `@CRV` to `com.google.common`, the Google-internal cleanups found at least one case of `equals` where someone was meant to write `assertTrue(x.equals(y))` (and a similar case of `containsEntry`). We found some cases even for things like `ImmutableList.of` and , where people meant to write `return ImmutableList.of(...)` (inside an `if()`). These weren't super common, but I would expect a few similar finds if we turned on checking for more methods.
